### PR TITLE
[Sprint 4] Data Screen + Functionality to Delete Specific Data

### PR DESCRIPTION
**Pull request by maxbalves**
## Description
- A new screen allows users to see all their likes, dislikes, comments, and posts.
- Users are able to delete specific likes, dislikes, comments, and posts by swiping to the left.

## Milestones
Milestone/Sprint 4, [#30 Implement screen to see posts, likes, dislikes](https://github.com/maxbalves/PostBy/issues/30)

## Test Plan
- Tested the deletion process with all data types (Likes/Dislikes/Comments/Posts) and various possibilities:
    - More than 10+ posts, delete 2, and try to add 2 more to see if they display
    - Check that infinite scrolling works even after deletion
    - Check if relations are deleted properly
    - Check if comments are deleted after a post is deleted
    - etc...

_Demo:_

## Future Plans
- I hope to work on some of the UI in order to allow the app to be more user friendly. The goal is to improve the design of every screen, including the Data Screen just created.

### CODE REVIEW

**Comment by maxbalves** [comment]
Discovered another error with likes/dislikes relations leaving hidden data behind after a post is deleted. Working on the fix.

File: PostBy/View Models/PostViewModel.m [hunk 1]
```diff
@@ -269,9 +269,20 @@ - (void) deletePost {
...
+        // Remove comment from author's relation to prevent hidden data left behind
+        PFUser *author = comment[AUTHOR_FIELD];
+        PFRelation *authorCommentsRelation = [author relationForKey:COMMENTS_RELATION];
+        [authorCommentsRelation removeObject:comment];
+        [author saveInBackground];
+        
         [comment delete];
     }
     
+    // Delete Post from currentUser's relation otherwise Parse will leave hidden data behind
```
Comment (raozhang-fb) [inline PostBy/View Models/PostViewModel.m:281]: 👍

File: PostBy/View Controllers/SettingsViewController.m [hunk 2]
```diff
@@ -317,4 +318,48 @@ - (void) logoutUser {
...
+    NSDictionary *data = @{
+        @"navTitle" : @"Dislikes",
+        @"relation" : DISLIKES_RELATION
+    };
+    
+    [self performSegueWithIdentifier:@"SettingsShowData" sender:data];
+}
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    if ([segue.identifier isEqualToString:@"SettingsShowData"]) {
```
Comment (raozhang-fb) [inline PostBy/View Controllers/SettingsViewController.m:358]: Can this be assigned with a variable or constant? "SettingsShowData"

File: PostBy/View Controllers/DataViewController.m [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    }
+    
+    return cell;
+}
+
+- (BOOL) shouldRelationShowPosts:(NSString *)relation {
+    BOOL isLikesRelation = [relation isEqualToString:LIKES_RELATION];
+    BOOL isDislikesRelation = [relation isEqualToString:DISLIKES_RELATION];
+    BOOL isPostsRelation = [relation isEqualToString:POSTS_RELATION];
+    return (isLikesRelation || isDislikesRelation || isPostsRelation);
```
Comment (raozhang-fb) [inline PostBy/View Controllers/DataViewController.m:93]: Well formatted condition checking. 👍

File: PostBy/View Controllers/DataViewController.m [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+- (BOOL) shouldRelationShowComments:(NSString *)relation {
+    return ([relation isEqualToString:COMMENTS_RELATION]);
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.arrayOfData.count;
+}
+
+// Infinite scrolling
```
Comment (raozhang-fb) [inline PostBy/View Controllers/DataViewController.m:104]: I am thinking whether the similar functionalities between data view and home view can be reused/shared. 🤔

**Review by raozhang-fb — COMMENTED** [review]
Great job! 🎉

**Pull request merged**